core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -190,7 +190,7 @@ private DeciderOutcome decide(final Workflow workflow, List<Task> preScheduledTa
             if (!pendingTask.isExecuted() && !pendingTask.isRetried() && pendingTask.getStatus().isTerminal()) {
                 pendingTask.setExecuted(true);
                 List<Task> nextTasks = getNextTask(workflow, pendingTask);
-                if (pendingTask.isLoopOverTask() && !nextTasks.isEmpty()) {
+                if (pendingTask.isLoopOverTask() && !TaskType.DO_WHILE.name().equals(pendingTask.getTaskType()) && !nextTasks.isEmpty()) {
```

Comment:
Since we have changed the way next task will be returned for last loopOver task we just need to check that in case of two sequential DO_WHILE task we should not change the task reference name of second DO_WHILE task when the first one gets completed.